source/m2mbase.cpp

```diff
@@ -247,14 +306,24 @@ uint16_t M2MBase::instance_id() const
     return _instance_id;
 }
 
-const String& M2MBase::interface_description() const
+String M2MBase::interface_description() const
 {
-    return _interface_description;
+    if(_interface_description!=NULL) {
+        return String ((const char*)_interface_description);
+        }
+    else {
+        return String("");
+    }
 }
 
-const String& M2MBase::resource_type() const
+String M2MBase::resource_type() const
 {
-    return _resource_type;
+    if(_resource_type!=NULL) {
+        return String ((const char*)_resource_type);
+        }
+    else {
+        return String("");
+    }
```

Comment:
Instead of copypasting same sequence for empty string handling, would it better to add some utility method to String class to handle it. Unfortunately the API allows the pointer to be NULL as allocation errors are mostly ignored, so this needs to be handled everywhere.